src/pq_flash_index.cpp

```diff
@@ -1181,13 +1243,27 @@ void PQFlashIndex<T, LabelT>::cached_beam_search(const T *query1, const uint64_t
             }
         }
     }
-    else if (_filter_to_medoid_id.find(filter_label) != _filter_to_medoid_id.end())
-    {
-        best_medoid = _filter_to_medoid_id[filter_label];
-    }
     else
     {
-        throw ANNException("Cannot find medoid for specified filter.", -1, __FUNCSIG__, __FILE__, __LINE__);
+        if (_filter_to_medoid_ids.find(filter_label) != _filter_to_medoid_ids.end())
+        {
+            for (uint64_t cur_m = 0; cur_m < _filter_to_medoid_ids[filter_label].size(); cur_m++)
```

Comment:
use "const auto& medio_ids = _filter_to_medoid_ids[filter_label];" to refer to the vector to avoid lookuping _filter_to_medoid_ids map multiple times

Reply:
the same for line 1254 and line 1258

Reply:
done